src/main/scala/com/fulcrumgenomics/bam/Bams.scala

```diff
@@ -219,6 +223,65 @@ object Bams extends LazyLogging {
     }
   }
 
+  /** Returns an iterator over the records in the given reader such that the order of the records returned is determined
+    * by the value of the given SAM tag, which can optionally be transformed.
```

Comment:
Do you want to enforce any sub-ordering?  E.g. by queryname and then flags?  Or do you think that sorting by just tag is sufficient?

Reply:
For now it is sufficient, we can add that in later (without harm) if we want it.  I don't see anywhere that sorting is stable, but if it was we could first sort by queryname/flags then by tag, but that's ugly.